Lovely ui

nmohamed opened the pull request:
-need to add general listview animation for making the listview animate out when you press the button
-general animation stuff
-make sure listview history works with non-reddit responses
-add overscroll things

-need to pull from master, apparently it thinks i'm up to date even in git shell so I'm mainly PRing to fix this and get help with animations, and also code reviewing

nmohamed committed:
added listview and custom listviewadapter class

-it doesn't crash

nmohamed committed:
stuff i've done so far - added footer and class

added a footer

nmohamed committed:
commit to save. heights of listview/listelement

commit to save what i've done. right now, i'm trying to get the heights
of the last item in the listview and the listview height (the former
doesnt seem correct for some reason, the later is probably okay... need
to test it somehow.)

nmohamed committed:
committing for help

nmohamed committed:
get sizes and updates the footer.

size is not exact but it kind of works

nmohamed committed:
working history UI (mostly)

-doesn't work well for error texts
-started adding animations, but they dont work

nmohamed committed:
fixing code and adding animations

-need to properly show error/notfound responses with history
-need to make listview out animation work

YuzhongHuang commented:
Probably needs to fix the merge confilct first?

nmohamed commented:
yup, read my comment above

YuzhongHuang commented on app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/SpeechFragment.java:
```diff
@@ -159,51 +182,85 @@ public void speechResultCallback(ArrayList voiceResult) {
         }
     }
 
+    public void showComment(String comment){ //a comment will always show, so do it here
```
I think it is because the master has not been merged to that branch. If there is no voice input, the app will will crash because voiceInput.get(0) is null.

Also, voiceInput.get(0) is a string itself, so no need to call .toString

YuzhongHuang commented on app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/SpeechFragment.java:
```diff
@@ -159,51 +182,85 @@ public void speechResultCallback(ArrayList voiceResult) {
+        listViewAdapter.notifyDataSetChanged();
+
+        //change footer view
+        if(firstResponse == true) {
```
can be simplified to if (firstResponse)

nmohamed committed:
resolving merge conflicts and integrating UI

nmohamed committed:
add animation

nmohamed committed:
solving merge conflicts

davidabrahams commented on app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/ListViewAdapter.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    @Override public Object getItem(int position) {
+        ArrayList<String> items = new ArrayList<>(2);
```
why initialize to length 2?

davidabrahams commented on app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/ListViewAdapter.java:
```diff
@@ -0,0 +1,119 @@
+        LayoutInflater inflater = LayoutInflater.from(context);
+
+        if (convertView == null) {
+            convertView = inflater.inflate(R.layout.list_item, null);
```
why are you passing null here?

nmohamed commented:
things to do: 
-make edit text wrap around
-see if you can get rid of edit text lag
-make listview not viewable after getting out of type mode
-make type mode a one click thing

nmohamed commented:
things to do as final edits:
-put pre-generated responses into strings.xml

nmohamed committed:
fixed edit text

davidabrahams commented on app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/SpeechFragment.java:
```diff
@@ -17,86 +17,86 @@
             Arrays.asList(
                     "You know, life without Wi-Fi is hard.",
-                    "maybe you should move to Canada, we have really good internet here :o)",
+                    "maybe you should move to canada, we have really good internet here :o)",
```
capitalize?

nmohamed commented:
bugs:
-it doesnt speak. seems like going to the info fragment turns this on or off.

LucyWilcox committed:
Added strings

nmohamed committed:
better edittext

now it's one click

LucyWilcox committed:
Added things to string and info page

LucyWilcox committed:
Added one last thing

nmohamed committed:
changed edit text to word wrap and char limit

LucyWilcox committed:
Changed App Name

LucyWilcox committed:
Merge remote-tracking branch 'refs/remotes/origin/lovely-ui' into adding-strings

# Conflicts:

#	app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/SpeechFragment.java

LucyWilcox committed:
Don't need onCreate in InfoFragment

Commit:
fix the mute problem

nmohamed commented:
I have a question - when you get back the speech result, it searches the exact string. The way the edit text works is that it will only take 100 characters, and so your response will be cut off at that point. Right now it still searches the potentially >100 characters `voiceInput`, but instead we could make it search the <100 characters `editText.getText()` (if you `editText.setText` it'll automatically cut off at 100 characters). Thoughts?

nmohamed committed:
many small changes

-changed launcher
-made mic images smaller
-got rid of overscroll
-changed edittext character limit
-cleaned up/renamed variables
-changed color

LucyWilcox committed:
Updated wifi check

LucyWilcox committed:
Merge remote-tracking branch 'refs/remotes/origin/lovely-ui' into adding-strings

# Conflicts:

#	app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/SpeechFragment.java

LucyWilcox committed:
Mute bug fixed, wifi working in all cases

LucyWilcox committed:
Changed "false" to null

nmohamed committed:
Merge pull request #61 from greneroom/adding-strings

Adding strings

Commit:
fix the info page bug

LucyWilcox committed:
Cleaned some code and fixed last wifi issue

Commit:
fix the info page not speaking bug

nmohamed committed:
EditText "back" functionality

Made pressing the back button after the keyboard for the edittext is up
search your response

nmohamed committed:
Merge pull request #63 from greneroom/no-wifi-type

Cleaned some code and fixed last wifi issue

Commit:
clean the debuging code

LucyWilcox committed:
Merge pull request #64 from greneroom/info-error

Info error

nmohamed committed:
Merge remote-tracking branch 'origin/lovely-ui' into lovely-ui

# Conflicts:

#	app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/SpeechFragment.java

nmohamed committed:
Merge remote-tracking branch 'origin/lovely-ui' into lovely-ui

# Conflicts:

#	app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/SpeechFragment.java

Commit:
fix the back-button-error

Commit:
Merge branch 'lovely-ui' of github.com:greneroom/Hi-Reddit into back-button-error

nmohamed committed:
tiny changes

just formatting

Commit:
Merge pull request #68 from greneroom/tiny-changes-lovely-ui

tiny changes

Commit:
tiny changes

LucyWilcox committed:
Changed Toast to commentcallback

LucyWilcox committed:
Fixed long hold on empty space

nmohamed committed:
try saying "hi reddit" ;-)

LucyWilcox committed:
Deleted comment

nmohamed committed:
merging

nmohamed committed:
Merge branch 'hi-reddit-response' of https://github.com/greneroom/Hi-Reddit into hi-reddit-response

davidabrahams commented on app/src/main/java/com/mobileproto/hireddit/hireddit/speech/WordToSpeech.java:
```diff
@@ -46,8 +46,12 @@ public void stop() {
 
-    public void flipMute() {
-        mute = !mute;
+    public void Mute() {
```
uncapitalize

nmohamed committed:
Merge pull request #67 from greneroom/back-button-error

fix the back-button-error

nmohamed commented:
also wow, I didn't realize we had 41 files to change...

greneroom commented:
If it's an easy fix than sure. Otherwise, it's not a big deal

On Fri, Dec 18, 2015 at 12:22 AM Nora Mohamed <notifications@github.com>
wrote:

> also wow, I didn't realize we had 41 files to change...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/greneroom/Hi-Reddit/pull/56#issuecomment-165675910>.
>

LucyWilcox committed:
Merge pull request #70 from greneroom/hi-reddit-response

try saying "hi reddit" ;-)

LucyWilcox commented:
I agree with David, we can do that, but if not that's fine. Also David do you have things to merge into this branch? Or are we all good?

LucyWilcox commented on app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/ListViewAdapter.java:
```diff
@@ -0,0 +1,115 @@
+//        } else {
+//            Animation historyAnimation = AnimationUtils.loadAnimation(context, R.anim.down_from_top);
+//            convertView.startAnimation(historyAnimation);
+//        }
```
Should we get rid of these commented sections and space?

nmohamed replied:
Probably. The animation was pretty cool though, wish it worked.

LucyWilcox committed:
Made mute lowercase

nmohamed commented:
Okay, the reason I was asking about it is because you could potentially be more specific with what you search, but it'll only show the cut-off message. So, say I searched "i like dogs and cats" and it cut off at "i like dogs", but the comment you get was searched using the "i like dogs and cats" string. You get "i hate dogs and like cats" despite it not showing that you searched for cats. So, should we make the displayed search query representative of what it actually searched? Or should we allow them to search whatever they want in the case that they speak outloud and search it, despite showing them something that's shorter?
It's a 1 or 2 line change.

LucyWilcox commented:
Oh, I think we should show everything they say if we are searching everything they say.

nmohamed commented:
But right now there's a cut off (which is necessary for making the editText work) at 100 characters. So, it currently will search everything you said despite showing not everything you said.

LucyWilcox commented:
So currently it cuts you off? And you're thinking of switching it to not cut off?

greneroom commented:
I don't have things to merge into this branch
On Fri, Dec 18, 2015 at 11:51 AM Lucy Wilcox <notifications@github.com>
wrote:

> So currently it cuts you off? And you're thinking of switching it to not
> cut off?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/greneroom/Hi-Reddit/pull/56#issuecomment-165835687>.
>

nmohamed commented:
No, I'm thinking of changing what gets sent to the reddit searcher - currently it gets the non-cut off version, and i'm asking if we should give it the cut-off version

LucyWilcox commented:
That might make sense: a higher chance of getting something from Reddit I think. Either way is fine though.

LucyWilcox committed:
Cleaning

LucyWilcox committed:
More cleaning

nmohamed committed:
cleaning code

deleted unused animations and cleaning comments and unused variables

nmohamed committed:
Merge remote-tracking branch 'refs/remotes/origin/hi-reddit-response' into lovely-ui

nmohamed committed:
Merge branch 'lovely-ui' of https://github.com/greneroom/Hi-Reddit into lovely-ui

LucyWilcox committed:
Changed bottom padding

nmohamed committed:
changing cut-off issue

nmohamed committed:
changed snoo's head

LucyWilcox committed:
Cleaning

nmohamed committed:
deleting unused images

nmohamed committed:
lil' edit
